### ISSUE TITLE
Volume mounts not working after upgrade

### ISSUE THREAD

**Issue by JoshKeegan** [docker/for-win#6680]
- [x] I have tried with the latest version of my channel (**Stable** or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: E3E44565-D1FE-46AD-BE6F-6A274FE27BF1/20200513104402

### Information

  - Windows Version: Windows 10 v1909 (Build 18363.778)
  - Docker Desktop Version: 2.3.0.2 (45183) - latest stable
  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM: no

I had a container set to always restart before upgrading to the latest docker desktop that had volume mounts to my local windows Z:\ drive. It was docker-compose and here's the compose file:
```yaml
version: '3.5'

services:
  mssql:
    image: mcr.microsoft.com/mssql/server:2019-latest
    environment:
      - ACCEPT_EULA=Y
      - MSSQL_AGENT_ENABLED=True
      - SA_PASSWORD=redacted
    volumes:
      - mssqlData:/var/opt/mssql
      - mssqlBackups:/backups
    restart: always
    ports:
      - 1433:1433

volumes:
  mssqlData:
  mssqlBackups:
    driver_opts:
        type: none
        device: /z/Backups/SQL Server
        o: bind
```

It no longer starts and fails with the following error when I try and start it manually:
```bash
Starting mssql_mssql_1 ... error

ERROR: for mssql_mssql_1  Cannot start service mssql: error while mounting volume '/var/lib/docker/volumes/mssql_mssqlBackups/_data': failed to mount local volume: mount /z/Backups/SQL Server:/var/lib/docker/volumes/mssql_mssqlBackups/_data, flags: 0x1000: no such file or directory

ERROR: for mssql  Cannot start service mssql: error while mounting volume '/var/lib/docker/volumes/mssql_mssqlBackups/_data': failed to mount local volume: mount /z/Backups/SQL Server:/var/lib/docker/volumes/mssql_mssqlBackups/_data, flags: 0x1000: no such file or directory
Encountered errors while bringing up the project.
```

The Z:\ drive is still shared, and I have tried removing it, restarting docker & readding it to no avail:
![image](https://user-images.githubusercontent.com/1682204/81803708-fe329f80-950f-11ea-8948-dbba2c6cfed0.png)

When I try and mount the Z:\ drive to another container it appears to still work as expected though (e.g. via `docker run --rm -it -v Z:/:/z debian`).

### Steps to reproduce the behavior

I don't know as it was an existing container from before the upgrade.
The only odd behaviour I found when trying to reproduce was that mounting two drives at the same time failed, e.g. 
```ps
docker run --rm -it -v C:/:/c -v Z:/:z debian
```
errors with:
```ps
C:\Program Files\Docker\Docker\resources\bin\docker.exe: Error response from daemon: invalid volume specification: '/host_mnt/z:z'.
See 'C:\Program Files\Docker\Docker\resources\bin\docker.exe run --help'.
```
Mounting either of the drives individually works as expected, and so does multiple mounts on the same drive.

**Comment by Atrejoe** [comment]
This duplicates #6646

**Comment by JoshKeegan** [comment]
@Atrejoe I am not sure if they are the same issue. In my case, the container won't even start:
```bash
ERROR: for mssql_mssql_1  Cannot start service mssql: error while mounting volume '/var/lib/docker/volumes/mssql_mssqlBackups/_data': failed to mount local volume: mount /z/Backups/SQL Server:/var/lib/docker/volumes/mssql_mssqlBackups/_data, flags: 0x1000: no such file or directory
```
In that issue it appears to start the container and then the application crashes.

**Comment by sdedalus** [comment]
We are experiencing the same issue.

**Comment by djg2002** [comment]
I’m seeing a similar thin - see https://github.com/docker/for-win/issues/6700

**Comment by mat007** [comment]
@JoshKeegan this works fine for me from a cmd shell:
```
docker run --rm -it -v C:/:/c -v D:/:d debian
```
(I don’t have a Z drive 😁)

What shell are you using?

**Comment by mat007** [comment]
@JoshKeegan I think your initial issue may actually be similar to #6285
Does it work if you change you compose file to do
```
        device: /host_mnt/z/Backups/SQL Server
```
?

**Comment by JoshKeegan** [comment]
> 
> 
> @JoshKeegan this works fine for me from a cmd shell:
> 
> ```
> docker run --rm -it -v C:/:/c -v D:/:/d debian
> ```
> 
> (I don’t have a Z drive 😁)
> 
> What shell are you using?

I was just running that in PowerShell. 
I've downgraded to 2.2.0.5 for now so can't try again to see whether the drive letter was causing the issue (but have checked and my original example works now I've downgraded).



> 
> 
> @JoshKeegan I think your initial issue may actually be similar to #6285
> Does it work if you change you compose file to do
> 
> ```
>         device: /host_mnt/z/Backups/SQL Server
> ```
> 
> ?

That does look similar. I can't try doing the modification now due to having downgraded, but I do remember inspecting the containers and seeing some with /host_mnt at the beginning and some without. It would make sense if the ones with /host_mnt were the containers I'd added since the upgrade.

**Comment by mat007** [comment]
Ah I just realized what was wrong with your
```
docker run --rm -it -v C:/:/c -v Z:/:z debian
```
It’s missing a `/` before the `z`.
Sorry I didn’t pay attention when typing my own command, if I remove the `/` I get the same error as you did
```
# docker run --rm -it -v C:/:/c -v D:/:d debian
docker: Error response from daemon: invalid volume specification: '/host_mnt/d:d': invalid mount config for type "bind": invalid mount path: 'd' mount path must be absolute.
See 'docker run --help'.
```

**Comment by mat007** [comment]
We now have a private build with a fix, if you would like to try: https://download-stage.docker.com/win/stable/45279/Docker%20Desktop%20Installer.exe
We will also make a new release in very likely a few days.
Thanks!

**Comment by JoshKeegan** [comment]
> It’s missing a `/` before the `z`.
Ahh, good spot!
This issue now looks the same as the other one you linked, so I'm guessing that private build will help me :)

I will try and give that release a go at some point, but I haven't recreated the SQL Server cpntainer yet so will need to find time to do that first :)

**Comment by TobiasCortese** [comment]
@mat007 

I am unable to mount a volume with the latest stable build (2.3.0.2) which I believe is related to this issue. The problem is fixed in the Private Build you posted on May 15 but it is also fixed in the latest Edge Build 2.3.1.0. The Edge build is preferred because it's captured within the release process.

In general, it's unsettling that the issue was first unidentified within a day of the release of the latest stable build (May 11) but Docker has yet to release a stable build update with the problem fixed and the issue doesn't appear to be listed in the known issues for the current stable build. 

So, to ease my mind and instill confidence, could you please explain the delay and the lack of communication?  

Here is my scenario, which doesn't mount with the current Stable build but does mount with the current Edge build...

Dockerfile
FROM continuumio/anaconda3
EXPOSE 48080
COPY requirements.txt /
RUN pip install -r /requirements.txt
CMD ["jupyter", "lab", "--ip='0.0.0.0'", "--port=48080", "--no-browser", "--allow-root"]

$ docker build -t jupyterlab:latest .
$ docker run --name jupyterlab --rm -p 48080:48080 -v "/$(pwd)"://home/ jupyterlab

**Comment by stephen-turner** [comment]
We are planning a stable patch release this week. In general we try not to do a patch release until we've bundled up several issues, because our users don't like to update too often. That's why we post private builds on tickets, to unblock the people who have seen a particular issue while not bombarding everyone else with updates.

**Comment by Gemorroj** [comment]
the build `2.3.0.3` didn't work for me. it worked for one day and broke down again.

**Comment by mat007** [comment]
A fix has now been released in Docker Desktop Stable 3.2.0.3, I’ll go ahead and close this issue.
If you’re still seeing the same error (or a similar one) after upgrading to 3.2.0.3, could you please open a fresh issue?
Thanks!
